src/main/java/events/ImportUserEventsGcs.java

```diff
@@ -0,0 +1,135 @@
+
+  private static final String GCS_ERRORS_BUCKET = String.format("%s/error",
+      GCS_BUCKET);
+
+  private static final String GCS_EVENTS_OBJECT = "user_events.json";
+
+  /*
+  TO CHECK ERROR HANDLING USE THE JSON WITH INVALID USER EVENT
+  GCS_EVENTS_OBJECT = "user_events_some_invalid.json"
+   */
+
+  // get user events service client
+  private static UserEventServiceClient getUserEventsServiceClient()
+      throws IOException {
+    UserEventServiceSettings userEventServiceSettings = UserEventServiceSettings.newBuilder()
+        .setEndpoint(ENDPOINT)
+        .build();
+    return UserEventServiceClient.create(userEventServiceSettings);
+  }
+
+  // get import user events from gcs request
+  public static ImportUserEventsRequest getImportEventsGcsRequest(
+      String gcsObjectName) {
+    // TO CHECK ERROR HANDLING PASTE THE INVALID CATALOG NAME HERE:
+    // DEFAULT_CATALOG = "invalid_catalog_name"
+
+    GcsSource gcsSource = GcsSource.newBuilder()
+        .addInputUris(String.format("%s/%s", GCS_BUCKET, gcsObjectName))
+        .build();
+
+    UserEventInputConfig inputConfig = UserEventInputConfig.newBuilder()
+        .setGcsSource(gcsSource)
+        .build();
+
+    ImportErrorsConfig errorsConfig = ImportErrorsConfig.newBuilder()
+        .setGcsPrefix(GCS_ERRORS_BUCKET)
+        .build();
+
+    ImportUserEventsRequest importRequest = ImportUserEventsRequest.newBuilder()
+        .setParent(DEFAULT_CATALOG)
+        .setInputConfig(inputConfig)
+        .setErrorsConfig(errorsConfig)
+        .build();
+
+    System.out.printf(
+        "Import user events from google cloud source request: %s%n",
+        importRequest);
+
+    return importRequest;
+  }
+
+  // call the Retail API to import user events
+  public static void importUserEventsFromGcs()
+      throws IOException, ExecutionException, InterruptedException {
+    ImportUserEventsRequest importGcsRequest = getImportEventsGcsRequest(
+        GCS_EVENTS_OBJECT);
+
+    OperationFuture<ImportUserEventsResponse, ImportMetadata> gcsOperation
+        = getUserEventsServiceClient().importUserEventsAsync(importGcsRequest);
+
+    System.out.printf("The operation was started: %s%n",
+        gcsOperation.getName());
+
+    while (!gcsOperation.isDone()) {
+      System.out.println("Please wait till operation is done.");
+
+      getUserEventsServiceClient().awaitTermination(30, TimeUnit.SECONDS);
+
+      System.out.println("Import user events operation is done.");
+
+      System.out.printf("Number of successfully imported events: %s%n",
+          gcsOperation.getMetadata().get().getSuccessCount());
+
+      System.out.printf("Number of failures during the importing: %s%n",
+          gcsOperation.getMetadata().get().getFailureCount());
+
+      System.out.printf("Operation result: %s%n", gcsOperation.get());
+    }
+  }
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException {
+    importUserEventsFromGcs();
```

Comment:
I have got an error

`The operation was started: projects/945579214386/locations/global/catalogs/default_catalog/operations/import-user-events-4986265658792238492
Please wait till operation is done.
Import user events operation is done.
Number of successfully imported events: 4
Number of failures during the importing: 0
Operation result: errors_config {
  gcs_prefix: "gs://crs-interactive-tutorials/error"
}
import_summary {
  joined_events_count: 4
}

[WARNING] thread Thread[Gax-1,5,events.ImportUserEventsGcs] was interrupted but is still alive after waiting at least 14998msecs
[WARNING] thread Thread[Gax-1,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-1,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-0,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-2,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-2,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-1,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-3,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-2,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-4,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-3,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-4,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-5,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-5,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-6,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-7,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-8,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-9,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-10,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-11,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-12,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-13,5,events.ImportUserEventsGcs] will linger despite being asked to die via interruption
[WARNING] NOTE: 21 thread(s) did not finish despite being asked to  via interruption. This is not a problem with exec:java, it is a problem with the running code. Although not serious, it should be remedied.
[WARNING] Couldn't destroy threadgroup org.codehaus.mojo.exec.ExecJavaMojo$IsolatedThreadGroup[name=events.ImportUserEventsGcs,maxpri=10]
java.lang.IllegalThreadStateException
        at java.lang.ThreadGroup.destroy(ThreadGroup.java:778)
        at org.codehaus.mojo.exec.ExecJavaMojo.execute(ExecJavaMojo.java:293)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:08 min
[INFO] Finished at: 2022-01-04T17:01:44+01:00
[INFO] Final Memory: 38M/734M
`